rust/core-lib/src/file.rs

```diff
@@ -172,6 +190,79 @@ impl FileManager {
         }
         Ok(())
     }
+
+    #[cfg(feature = "notify")]
+    pub fn update_current_position_in_tail(
+        &mut self,
+        buffer_id: BufferId,
+    ) -> Result<bool, FileError> {
+        if let Some(file_info) = self.file_info.get_mut(&buffer_id) {
+            let path = &file_info.path.to_owned();
+
+            let mut f = File::open(path).map_err(|e| FileError::Io(e, path.to_owned()))?;
+            let end_position =
+                f.seek(SeekFrom::End(0)).map_err(|e| FileError::Io(e, path.to_owned()))?;
+
+            file_info.tail_position = Some(end_position);
+            self.watcher.unwatch(path, OPEN_FILE_EVENT_TOKEN);
+            self.watcher.watch(path, false, OPEN_FILE_TAIL_EVENT_TOKEN);
+            Ok(true)
+        } else {
+            info!("Non-existent file cannot be tailed!");
+            Ok(false)
+        }
+    }
+
+    #[cfg(feature = "notify")]
+    pub fn disable_tailing(&mut self, buffer_id: BufferId) -> bool {
+        if let Some(file_info) = self.file_info.get_mut(&buffer_id) {
+            let path = &file_info.path.to_owned();
+            file_info.tail_position = None;
+            self.watcher.unwatch(path, OPEN_FILE_TAIL_EVENT_TOKEN);
+            self.watcher.watch(path, false, OPEN_FILE_EVENT_TOKEN);
+            true
+        } else {
+            info!("Non-existent file cannot be tailed!");
+            false
+        }
+    }
+}
+
+/// When tailing a file, instead of reading file from beginning, we need to get changes from the end.
+/// This method does that.
+#[cfg(feature = "notify")]
+fn try_tailing_file<P>(path: P, file_info: &FileInfo) -> Result<(Rope, FileInfo), FileError>
+where
+    P: AsRef<Path>,
+{
+    let mut f =
+        File::open(path.as_ref()).map_err(|e| FileError::Io(e, path.as_ref().to_owned()))?;
+    let end_position =
+        f.seek(SeekFrom::End(0)).map_err(|e| FileError::Io(e, path.as_ref().to_owned()))?;
+
+    let current_position = file_info.tail_position.expect(
+        "tail_position is None. Since we are tailing a file, tail_position cannot be None.",
+    );
+    let diff = end_position - current_position;
+    let mut buf = vec![0; diff as usize];
+    f.seek(SeekFrom::Current(-(buf.len() as i64))).unwrap();
+    f.read_exact(&mut buf).unwrap();
+
+    let mod_time =
+        f.metadata().map_err(|e| FileError::Io(e, path.as_ref().to_owned()))?.modified().ok();
```

Comment:
```suggestion
        f.metadata().map_err(|e| FileError::Io(e, path.clone()))?.modified().ok();
```

Reply:
(Not tested though since this currently doesn't apply to master)

Reply:
Will make this change and test it.

Reply:
Gives error
```
error[E0599]: no method named `clone` found for type `P` in the current scope
   --> core-lib/src/file.rs:252:56
    |
252 |         f.metadata().map_err(|e| FileError::Io(e, path.clone()))?.modified().ok();
    |                                                        ^^^^^ method not found in `P`
    |
    = help: items from traits can only be used if the type parameter is bounded by the trait
help: the following trait defines an item `clone`, perhaps you need to restrict type parameter `P` with it:
    |
234 | fn try_tailing_file<P: std::clone::Clone>(path: P, file_info: &FileInfo) -> Result<(Rope, FileInfo), FileError>
    |                     ^^^^^^^^^^^^^^^^^^^^

```

Reply:
Ah, fair enough then